### PR TITLE
V2 settings teams (Profil, Members, Appearance View)

### LINKED ISSUE #4240: 2.0 Settings / Team {View}
https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=1569%3A84778

### LINKED ISSUE #4343: /onboarding/step-4 - Upload button looks weird
Image

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Adds Profile, Members and Appearance page for teams settings

Fixes #4240
Fixes #4343 

Loom Video: https://www.loom.com/share/5315e1d1732440d4839b59d6d2dd47e4

**Environment**: Staging(main branch) / Production

### CODE REVIEW

File: packages/trpc/server/routers/viewer/teams.tsx [hunk 1]
```diff
@@ -194,14 +196,15 @@ export const viewerTeamsRouter = createProtectedRouter()
       memberId: z.number(),
     }),
     async resolve({ ctx, input }) {
-      if (!(await isTeamAdmin(ctx.user?.id, input.teamId))) throw new TRPCError({ code: "UNAUTHORIZED" });
+      const isAdmin = await isTeamAdmin(ctx.user?.id, input.teamId);
+      if (!isAdmin && ctx.user?.id !== input.memberId) throw new TRPCError({ code: "UNAUTHORIZED" });
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/teams.tsx:200]: Allow members to remove themselves from a team (by leaving the team)

File: packages/trpc/server/routers/viewer/teams.tsx [hunk 2]
```diff
@@ -449,7 +452,7 @@ export const viewerTeamsRouter = createProtectedRouter()
       dateTo: z.string(),
     }),
     async resolve({ ctx, input }) {
-      const team = await isTeamAdmin(ctx.user?.id, input.teamId);
+      const team = await isTeamMember(ctx.user?.id, input.teamId);
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/teams.tsx:455]: All members should be able to see the availability of other members

File: apps/web/components/v2/settings/teams/DisableTeamImpersonation.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { classNames } from "@calcom/lib";
```
Comment (joeauyeung) [inline apps/web/components/v2/settings/teams/DisableTeamImpersonation.tsx:1]: Had a call with @zomars and he mentioned we should move new components out of `apps/web/components` and into `packages/features`. Should @CarinaWolli do the same here?

File: packages/ui/v2/core/layouts/SettingsLayout.tsx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "@radix-ui/react-collapsible";
```
Comment (joeauyeung) [inline packages/ui/v2/core/layouts/SettingsLayout.tsx:1]: It was my understanding that I was handling this part of teams. I have #4347 open right now waiting for this PR. I see how you've done the routing. What's the best way to continue without causing too much merge conflicts?

File: apps/web/components/v2/settings/ImageUploader.tsx [hunk 5]
```diff
@@ -112,11 +112,9 @@ export default function ImageUploader({
         (opened) => !opened && setFile(null) // unset file on close
       }>
       <DialogTrigger asChild>
-        <div className="flex items-center">
```
Comment (CarinaWolli) [inline apps/web/components/v2/settings/ImageUploader.tsx:115]: Fixes #4343

File: packages/ui/v2/core/layouts/SettingsLayout.tsx (hunk 4 above)
Comment (CarinaWolli) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:1]: I merged your branch [`v2/settings-shell`](https://github.com/calcom/cal.com/tree/v2/settings-shell) into this branch to already connect the shell with my implemented views. I asked @zomars, he said it's best to open up a PR into `v2/settings-shell` branch. I think once your branch is merged, this should automatically set the PR to `main`

File: apps/web/components/v2/settings/teams/DisableTeamImpersonation.tsx (hunk 3 above)
Comment (zomars) [reply apps/web/components/v2/settings/teams/DisableTeamImpersonation.tsx:1]: Agreed. Will try to do it myself.

**Review by zomars — APPROVED** [review]
Ready to merge

LGTM

**Pull request merged**